File: app-android/src/main/java/org/mtransit/android/ui/view/common/IActivity.java
package org.mtransit.android.ui.view.common;

import android.app.Activity;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.LifecycleOwner;

import org.mtransit.android.common.IContext;

public interface IActivity extends IContext {

	@Nullable
	Activity getActivity();

	@NonNull
	Activity requireActivity() throws IllegalStateException;

	@NonNull
	LifecycleOwner getLifecycleOwner();

	void finish();
}

File: app-android/src/main/java/org/mtransit/android/ui/view/common/IActivity.kt
package org.mtransit.android.ui.view.common

import android.app.Activity
import androidx.fragment.app.Fragment
import androidx.lifecycle.LifecycleOwner
import org.mtransit.android.common.IContext
import java.lang.IllegalStateException

interface IActivity : IContext, LifecycleOwner {

    val activity: Activity?

    @Throws(IllegalStateException::class)
    fun requireActivity(): Activity

    val currentFragment: Fragment?

    fun finish()
}
